### DISCUSSION TITLE
Splitting steps into two phases causes issues with friction between surfaces

### DISCUSSION THREAD

**Discussion by nanonuq** [OpenFDEM-geomechanics/OpenFDEM.github.io#2] [Q&A] [answered]
I'm doing UCS simulation. I find that if I split the steps into two phases (e.g. resuming from a saved stressed condition), the sample stops being pinned where it touches the platens and relaxes. If I do the same amount of steps in one shot, the sample remains pinned.

This code replicates the issue, if you run it as-is then run it again with two 100,000 steps rather than 1x200,000 steps, you can see the x displacement is significantly different between the two samples at the end of the run. The sample stops being pinned when the second round of steps begins, despite no change in the boundary conditions.

What's going on and how can I get more consistent behaviour?

`of.new

of.set.omp 3

of.geometry.square 'rock' x 0 25.4e-3 y 0 50.8e-3
of.geometry.square 'lower_plate' x -5e-3 30.4e-3 y -5e-3 0
of.geometry.square 'upper_plate' x -5e-3 30.4e-3 y 50.8e-3 55.8e-3
of.geometry.mesh.size 'default' 0.7e-3
of.geometry.mesh delaunay

of.group.element 'rock_mesh' range box.in [0 25.4e-3 0 50.8e-3]

of.mat.element 'rock' ELASTIC den 2600 E 7.2e9 v 0.18 damp 0.6
of.mat.element 'upper_plate' ELASTIC den 2700 E 70e9 v 0.2 damp 0.9
of.mat.element 'lower_plate' ELASTIC den 2700 E 70e9 v 0.2 damp 0.9

of.mat.contact 'default' MC fric 0.269

of.group.nodal.from.element 'upper_plate' 'upper_plate'
of.group.nodal.from.element 'lower_plate' 'lower_plate'

of.boundary.nodal.velocity 'upper_plate' x 0 y -2e-2
of.boundary.nodal.velocity 'lower_plate' x 0 y 0e-2

of.set.result 'resumptionissue2'

of.history.pv.interval 10000
of.history.pv.field 'default'

of.step 200000
#of.step 100000
#of.step 100000
of.finalize

`

**Comment by jackgoodman2** [comment, marked as answer]
Hi, @nanonuq , thanks for your report.
I checked what's your statement and found that the ks of contact was recaclculated twice in the second runing. This bug is fixed now and i will push the new version shortly.

Thanks.

**Comment by pfefailed** [comment]
Please can you give us an idea how to get the source code ?
